### PR TITLE
Use json.RawMessage for AnsibleVars

### PR DESCRIPTION
**Pull request by rabi**
This changes AnsibleVars to a map, so that individual vars could be changed/patched.

Depends-On: https://github.com/openstack-k8s-operators/install_yamls/pull/442

### CODE REVIEW

**Comment by fao89** [comment]
> Build failed (check pipeline). For information on how to proceed, see http://docs.openstack.org/infra/manual/developers.html#automated-testing
> 
> https://review.rdoproject.org/zuul/buildset/c6b33e67386540d6a84c2ef12b3c44f4
> 
> heavy_check_mark [dataplane-operator-docs-preview ](https://review.rdoproject.org/zuul/build/ffd91d380f694ccdbeba81762b7f9517) SUCCESS in 1m 49s ✔️ [dataplane-operator-content-provider ](https://review.rdoproject.org/zuul/build/85892f083b8246468fee68d4bd509250) SUCCESS in 1h 03m 52s x [dataplane-operator-crc-podified-edpm-baremetal ](https://review.rdoproject.org/zuul/build/3dbb16c3038b41a48ebda67a6a104ead) **FAILURE** in 33m 00s x [dataplane-operator-crc-podified-edpm-deployment ](https://review.rdoproject.org/zuul/build/e4240034d4d84a72ba7f0050ac3f3429) **FAILURE** in 48m 13s

I believe we need install_yamls PR for making the CI green
https://github.com/openstack-k8s-operators/install_yamls/blob/main/scripts/gen-edpm-kustomize.sh#L113
@raukadah is it possible to use `Depends on` here?

**Comment by fao89** [comment]
I started this: https://github.com/openstack-k8s-operators/install_yamls/pull/437

**Comment by rabi** [comment]
> https://github.com/openstack-k8s-operators/dataplane-operator/pull/319

Thanks for trying that. This still needs local testing to confirm that the approach works in all scenarios. I already had one in-progress install_yamls PR that I have submitted with Depends-On.

**Comment by raukadah** [comment]
> > Build failed (check pipeline). For information on how to proceed, see http://docs.openstack.org/infra/manual/developers.html#automated-testing
> > https://review.rdoproject.org/zuul/buildset/c6b33e67386540d6a84c2ef12b3c44f4
> > heavy_check_mark [dataplane-operator-docs-preview ](https://review.rdoproject.org/zuul/build/ffd91d380f694ccdbeba81762b7f9517) SUCCESS in 1m 49s heavy_check_mark [dataplane-operator-content-provider ](https://review.rdoproject.org/zuul/build/85892f083b8246468fee68d4bd509250) SUCCESS in 1h 03m 52s x [dataplane-operator-crc-podified-edpm-baremetal ](https://review.rdoproject.org/zuul/build/3dbb16c3038b41a48ebda67a6a104ead) **FAILURE** in 33m 00s x [dataplane-operator-crc-podified-edpm-deployment ](https://review.rdoproject.org/zuul/build/e4240034d4d84a72ba7f0050ac3f3429) **FAILURE** in 48m 13s
> 
> I believe we need install_yamls PR for making the CI green https://github.com/openstack-k8s-operators/install_yamls/blob/main/scripts/gen-edpm-kustomize.sh#L113 @raukadah is it possible to use `Depends on` here?

Yes, Depends-On will work fine here. Either we can add Depends-On: <this patch> on install_yamls pr or vice versa will work.

**Comment by rabi** [comment]
recheck

**Comment by rabi** [comment]
recheck

**Comment by rabi** [comment]
recheck

**Comment by rabi** [comment]
recheck

File: api/v1beta1/common.go [hunk 1]
```diff
@@ -49,9 +50,10 @@ type NodeSection struct {
 	// AnsiblePort SSH port for Ansible connection
 	AnsiblePort int `json:"ansiblePort,omitempty"`
 
-	// +kubebuilder:validation:Optional
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:Schemaless
 	// AnsibleVars for configuring ansible
-	AnsibleVars string `json:"ansibleVars,omitempty"`
+	AnsibleVars map[string]json.RawMessage `json:"ansibleVars,omitempty"`
```
Comment (fao89) [inline api/v1beta1/common.go:56]: 👍🏽

File: config/samples/dataplane_v1beta1_openstackdataplane_baremetal.yaml [hunk 2]
```diff
@@ -53,7 +53,8 @@ spec:
         ansibleUser: cloud-admin
         ansiblePort: 22
         ansibleSSHPrivateKeySecret: dataplane-ansible-ssh-private-key-secret
-        ansibleVars: |
+        ansibleVars:
+          nova: {}
```
Comment (fao89) [inline config/samples/dataplane_v1beta1_openstackdataplane_baremetal.yaml:57]: isn't it supposed to be under `node`/`nodeTemplate`?
https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/api/v1beta1/common.go#L83

File: config/samples/dataplane_v1beta1_openstackdataplane_baremetal_with_ipam.yaml [hunk 3]
```diff
@@ -47,7 +47,8 @@ spec:
         managementNetwork: CtlPlane
         ansibleUser: cloud-admin
         ansiblePort: 22
-        ansibleVars: |
+        ansibleVars:
+          nova: {}
```
Comment (fao89) [inline config/samples/dataplane_v1beta1_openstackdataplane_baremetal_with_ipam.yaml:51]: ditto

File: config/samples/dataplane_v1beta1_openstackdataplane_baremetal.yaml (hunk 2 above)
Comment (rabi) [reply config/samples/dataplane_v1beta1_openstackdataplane_baremetal.yaml:57]: yeah, thanks for spotting!

**Review by fao89 — APPROVED** [review]
thank you!

**Comment by fao89** [comment]
recheck

**Comment by rabi** [comment]
We need to merge this and the `install_yamls` PR together to avoid broken CI. Anyway looks like edpm CI is failing with NODE_FAILUREs atm.

**Comment by fao89** [comment]
and bump dataplane-operator on openstack-operator

**Comment by rabi** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
recheck

**Review by slagle — APPROVED** [review]

**Pull request merged**